### PR TITLE
Use ligo binary from $PATH if available

### PR DESCRIPTION
**Pull request by callistonianembrace**
# 🌮 Taqueria PR

## 🪁 Description

`ligo` binary available globally can be faster to load and run. This PR looks into $PATH and if it finds it - uses it instead of the docker image

## 🪂 Pre-Merge Checklist (Definition of Done)

🚦 Required to merge:

- [x] ⛱️ I have completed this PR template in full and updated the title appropriately
- [x] ⛵ My code builds cleanly, and I have manually tested the changes
- [x] 🏄‍♂️ Another team member has built this branch and done manual testing on the change
- [ ] 🏖️ New and existing unit tests pass locally and in CI
- [ ] 🔱 The test plan has been implemented and verified by an SDET
- [ ] 🦀 Automated tests have been written and added to this PR
- [x] 🐬 I have commented my code, particularly in hard-to-understand areas
- [ ] 🤿 Corresponding changes have been made to all documentation
- [x] 🐚 Required changes to the VScE have been made
- [x] 🪸 Required updates to scaffolds have been made
- [x] 🚢 The release checklist has been completed

## 🛩️ Summary of Changes

This PR introduces a new function `baseDriverCmd` that tries to resolve a ligo binary from $PATH and uses it instead of the docker run command

## 🎢 Test Plan

_Please describe the testing strategy and plan for this PR. Keep this lightweight and anticipate any testing challenges_

## 🛸 Type of Change

- [ ] 🧽 Chore ➾
- [ ] 🛠️ Fix ➾
- [ ] ✨ Feature ➾
- [ ] 👷 Refactor ➾
- [ ] 🧪 Pre-Release ➾
- [ ] 🚀 Release ➾

### CODE REVIEW

**Review by ulrikstrid — COMMENTED** [review]
Initial pass looks good, I'm not sure if we should bump the version here or if it's done at release instead? It seems a bit noisy

**Comment by callistonianembrace** [comment]
>  I'm not sure if we should bump the version here or if it's done at release instead? 

That's Deno tooling I believe. It ran during the pre-commit hook and I had no control. I guess it was intentionally setup.

**Comment by mweichert** [comment]
Wow, thanks so much! I took a few minutes to review and it looks good thus far, but I'll need to do some testing and give a more thorough review a bit later. We'll be in touch. Thanks again!

File: taqueria-lib-ligo/common.ts [hunk 1]
```diff
@@ -77,3 +78,36 @@ export const configure = (dockerImage: string, dockerImageEnvVar: string) => ({
...
+	}
+}
+
+function getLigoBinaryFromPath() {
+	const { PATH } = process.env;
+	if (!PATH) {
+		return null;
+	}
+	const paths = PATH.split(delimiter);
+	for (const candidatePath in paths) {
```
Comment (mweichert) [inline taqueria-lib-ligo/common.ts:97]:
```suggestion
for (const candidatePath of paths) {
```

Otherwise, `candidatePath` will be the index of the array.

**Review by mweichert — CHANGES_REQUESTED** [review]
This looks great! I did find one typo that was preventing this PR from working as intended.

The only other things missing are:
- There's a ligo.ts file which provides the handler for the `ligo` task, used to execute arbitary ligo commands. Although this isn't properly needed when the `ligo` binary is available on the system, I think its more consistent to have it use the binary too if its available.
- A test in `tests/e2e/tests/e2e/ligo-plugin-e2e-tests.spec.ts` that assures the plugin will use the binary when available. Not a blocker for PR approval but is a blocker for release.

**Comment by mweichert** [comment]
> Initial pass looks good, I'm not sure if we should bump the version here or if it's done at release instead? It seems a bit noisy

In a PR, we probably shouldn't bump version #s. But that's ok. In the main branch, we use [CI in a XP-kind of way](https://www.youtube.com/watch?v=Sj0Ifupu6uA&t=76s), or at least moving closer in that direction.

**Comment by callistonianembrace** [comment]
@mweichert Can you please take look again? I have addressed feedback I believe.

**Comment by callistonianembrace** [comment]
I have added an e2e test too

File: taqueria-lib-ligo/common.ts [hunk 2]
```diff
@@ -1,79 +1,131 @@
...
-import { ProxyTaskArgs, RequestArgs } from '@taqueria/node-sdk/types';
-import { join } from 'path';
+import { getDockerImage, sendErr } from "@taqueria/node-sdk";
+import { ProxyTaskArgs, RequestArgs } from "@taqueria/node-sdk/types";
+import * as fs from "fs";
+import { delimiter, join } from "path";
 
 export interface LigoOpts extends ProxyTaskArgs.t {
-	command: string;
+  command: string;
```
Comment (callistonianembrace) [inline taqueria-lib-ligo/common.ts:7]: I'm not sure why the formatter is inconsistent here. Made sure I ran `dprint` finishing up

**Comment by mweichert** [comment]
@callistonianembrace I apologize, I've been ill as of late. I'll be sure to review your changes today. Thanks so much!

**Comment by zamrokk** [comment]
Really super fast now :P

**Pull request merged**